### ISSUE TITLE
Remove writing of default host.json

### ISSUE THREAD

**Issue by mathewc** [Azure/azure-functions-host#3508]
Commit https://github.com/Azure/azure-functions-host/commit recently added code that writes the default empty host.json file if one doesn't exist. We don't need to do that - the host can run w/o a physical file. In https://github.com/Azure/azure-functions-host/pull/2784 I made the change to allow running with no file, and removed the previous writing of an empty file.

**Comment by brettsam** [comment]
You're missing the commit link, but I'm guessing you were linking to this PR: https://github.com/Azure/azure-functions-host/pull/3320

@paulbatum, @fabiocav,  and I discussed this when I added the version requirement. We thought that auto-writing a file was a better step as it gave customers a good starting point to build from. It doesn't work in all scenarios, but we wanted to point people in the right direction when we could.

**Comment by mathewc** [comment]
@fabiocav @brettsam What do you guys think - should we keep the file write? My take is that it's an unnecessary file write that we can remove.

**Comment by paulbatum** [comment]
Yeah we added this back because we were worried about people not being able to get their host.json right (since they have to specify version:"2.0").

For example, lets say we revert this change, and you create a function app and then go to the section of the portal where you edit host.json. Will we generate an entry with version:"2.0"? Or will the user be required to figure this out themselves?

**Comment by paulbatum** [comment]
Similarly are there any local development flows that are assisted by the host generating this file?

**Comment by mathewc** [comment]
Good questions. Yes the portal currently uses the Kudu API `api/functions/config` to get the config, and when the file is missing, it returns `{}` empty (code [here](https://github.com/projectkudu/kudu/blob/f2f8e4c201682cd172c919c7c4e0c75dbca47833/Kudu.Core/Functions/FunctionManager.cs#L239)).

At this point since we've already added this back and may have established new dependencies on its presence, I agree its safest at this point to leave as is. The auto-generation of this file came up in a discussion today in the context of Powershell worker and I remembered this issue.
